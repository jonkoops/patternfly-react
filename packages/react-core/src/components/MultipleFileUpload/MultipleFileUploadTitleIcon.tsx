import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import styles from '@patternfly/react-styles/css/components/MultipleFileUpload/multiple-file-upload';
import { css } from '@patternfly/react-styles';
export interface MultipleFileUploadTitleIconProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside multiple file upload title icon */
  children?: ReactNode;
  /** Class to add to outer div */
  className?: string;
}

export const MultipleFileUploadTitleIcon: FunctionComponent<MultipleFileUploadTitleIconProps> = ({
  children,
  className,
  ...props
}: MultipleFileUploadTitleIconProps) => (
  <div className={css(styles.multipleFileUploadTitleIcon, className)} {...props}>
    {children}
  </div>
);

MultipleFileUploadTitleIcon.displayName = 'MultipleFileUploadTitleIcon';
