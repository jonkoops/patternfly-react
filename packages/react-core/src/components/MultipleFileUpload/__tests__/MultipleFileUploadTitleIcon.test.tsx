/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, test } from 'vitest';

import { MultipleFileUploadTitleIcon } from '../MultipleFileUploadTitleIcon';

describe('MultipleFileUploadTitleIcon', () => {
  test('renders with expected class names', () => {
    const { asFragment } = render(<MultipleFileUploadTitleIcon />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders custom class names', () => {
    const { asFragment } = render(<MultipleFileUploadTitleIcon className="test" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
