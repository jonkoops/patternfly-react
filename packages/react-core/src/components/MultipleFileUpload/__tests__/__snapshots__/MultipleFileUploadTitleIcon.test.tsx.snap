// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultipleFileUploadTitleIcon > renders custom class names 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-multiple-file-upload__title-icon test"
  />
</DocumentFragment>
`;

exports[`MultipleFileUploadTitleIcon > renders with expected class names 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-multiple-file-upload__title-icon"
  />
</DocumentFragment>
`;
